geoportal/c2cgeoportal_geoportal/scaffolds/create/build_tmpl

```
@@ -1,55 +1,76 @@
+parser.add_argument("--geoportal", action="store_true", help="Build only the geoportal image")
+parser.add_argument("--env", action="store_true", help="Force to create the `.env` file")
+parser.add_argument("--upgrade", help="Start upgrading the project to version")
+parser.add_argument("env_files", nargs="*", help="The environment files")
 
-CONFIG=true
-GEOPORTAL=true
-if [ "$1" == '--config' ]
-then
-    GEOPORTAL=false
-    shift
-fi
-if [ "$1" == '--geoportal' ]
-then
-    CONFIG=false
-    shift
-fi
+args = parser.parse_args()
 
-if [ $CONFIG == true ]
-then
-    set -x
-    docker build --tag=camptocamp/{{package}}-config \
-        --build-arg=PGSCHEMA=main .
-fi
-if [ $GEOPORTAL == true ]
-then
-    set -x
-    docker build --target=builder --tag=camptocamp/{{package}}-geoportal-dev \
-        --build-arg=PGSCHEMA=main \
-        --build-arg=GIT_HASH=$(git rev-parse HEAD) geoportal
-    docker build --tag=camptocamp/{{package}}-geoportal \
-        --build-arg=PGSCHEMA=main \
-        --build-arg=GIT_HASH=$(git rev-parse HEAD) geoportal
-fi
+if args.upgrade:
+    version_re = re.search(r"([0-9]\.[0-9])\.[0-9]", args.upgrade)
+    major_version = version_re.group(0) if version_re is not None else args.upgrade
+    full_version = args.upgrade if args.upgrade != "master" else "{{geomapfish_main_version}}"
+    with open("upgrade", "w") as f:
+        result = urllib.request.urlopen(
+            "https://raw.githubusercontent.com/camptocamp/c2cgeoportal/{}/scripts/upgrade".format(
+                major_version
+            )
+        )
+        if result.code != 200:
+            print("ERROR:")
+            print(result.read())
+            exit(1)
+        f.write(result.read().decode())
+    os.chmod("upgrade", os.stat("upgrade").st_mode | stat.S_IXUSR)
+    try:
+        subprocess.check_call(["./upgrade", full_version])
+    except subprocess.CalledProcessError:
+        exit(1)
```

Doesn't this mask the error ?

---

It masks the trace, but the real error message should be displayed by the upgrade command...